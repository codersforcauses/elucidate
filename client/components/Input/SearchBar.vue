<template>
  <div
    class="flex items-center h-10 bg-gray-200 rounded-md focus-within:outline focus-within:outline-2 focus-within:outline-gray-400"
  >
    <font-awesome-icon
      :icon="['fas', 'magnifying-glass']"
      class="mx-4 text-gray-400"
    />
    <input
      type="text"
      placeholder="Find a quiz"
      class="h-10 bg-transparent focus:outline-none w-80"
    />
  </div>
</template>

<script>
export default {
  name: 'SearchBar',
};
</script>
