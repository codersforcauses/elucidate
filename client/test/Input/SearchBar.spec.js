import SearchBar from '@/components/Input/SearchBar.vue';
import { mount } from '@vue/test-utils';

describe('SearchBar', () => {
  it('is a Vue instance', () => {
    const wrapper = mount(SearchBar);
    expect(wrapper.vm).toBeTruthy();
  });

  it('renders correctly', () => {
    const wrapper = mount(SearchBar);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('accepts text', () => {
    const wrapper = mount(SearchBar);
    wrapper.find('input').setValue('Test');
    expect(wrapper.find('input').element.value).toBe('Test');
  });
});
