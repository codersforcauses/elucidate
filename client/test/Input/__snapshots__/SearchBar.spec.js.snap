// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<div
  class="flex items-center h-10 bg-gray-200 rounded-md focus-within:outline focus-within:outline-2 focus-within:outline-gray-400"
>
  <font-awesome-icon
    class="mx-4 text-gray-400"
    icon="fas,magnifying-glass"
  />
   
  <input
    class="h-10 bg-transparent focus:outline-none w-80"
    placeholder="Find a quiz"
    type="text"
  />
</div>
`;
